rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -25,6 +25,54 @@
 #include <rosidl_typesupport_opensplice_cpp/message_type_support.h>
 #include <rosidl_typesupport_opensplice_cpp/service_type_support.h>
 
+inline std::string
+_create_type_name(
+  const message_type_support_callbacks_t * callbacks,
+  const std::string & sep)
+{
+  return std::string(callbacks->package_name) +
+         "::" + sep + "::dds_::" + callbacks->message_name + "_";
+}
+
+// *INDENT-OFF* (prevent uncrustify from using four space indention here)
+#define TRY_PLACEMENT_NEW(Destination, BufferForNew, FailureAction, Type, ...) try { \
```

Comment:
I should clarify I don't think `rmw` is the right place for this, since `rmw` is currently C only headers, but this would require C++ headers.

I guess it would be ok to add a C++ header to `rmw` under the guise of a "helper header". Maybe like `#include <rmw/impl/cpp/macros.hpp>`.